plugins/terraform/data_source_layer0_deploy.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/quintilesims/layer0/cli/client"
+)
+
+func dataSourceLayer0Deploy() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceLayer0DeployRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func datasourceLayer0DeployRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(client.Client)
+
+	deployName := d.Get("name").(string)
+	version := d.Get("version").(string)
+
+	deployID, err := resolveTags(client, deployName, map[string]string{
```

Comment:
Not sure I agree:
 - I feel it is more concise
 - with reference to the go-ism referenced in a later comment, params doesn't need to exist in the outside scope
 - but if we make type a fourth argument (as mentioned in a later comment), calling statement would look slightly better I suppose.